Detections/ASimWebSession/PotentiallyHarmfulFileTypes.yaml

```diff
@@ -16,45 +15,63 @@ requiredDataConnectors:
   - connectorId: Zscaler
     dataTypes:
       - CommonSecurityLog
-
 queryFrequency: 10m
 queryPeriod: 10m
 triggerOperator: gt
 triggerThreshold: 0
 tactics:
   - InitialAccess
+relevantTechniques:
+  - T1189
 query: |
-  let default_file_ext_blocklist = dynamic(['.ps1', '.vbs', '.bat', '.scr']);
-  let custom_file_ext_blocklist=toscalar(_GetWatchlist('RiskyFileTypes') | extend Extension=column_ifexists("Extension","") | where isnotempty(Extension) | summarize make_set(Extension));
+  let default_file_ext_blocklist = dynamic(['.ps1', '.vbs', '.bat', '.scr']); // Update this list as per your requirement
+  let custom_file_ext_blocklist=toscalar(_GetWatchlist('RiskyFileTypes')
+    | extend Extension=column_ifexists("Extension", "")
+    | where isnotempty(Extension)
+    | summarize make_set(Extension)); // If you have an extensive list, you can also create a Watchlist that includes the file extensions you want to detect
   let file_ext_blocklist = array_concat(default_file_ext_blocklist, custom_file_ext_blocklist);
-  _Im_WebSession(url_has_any=file_ext_blocklist, eventresult='Success')
-  | extend requestedFileName=tostring(split(tostring(parse_url(Url)["Path"]),'/')[-1])
-  | extend requestedFileExt=extract(@(\.\w+)$,1,requestedFileName, typeof(string))
+  _Im_WebSession(starttime=ago(10min), url_has_any=file_ext_blocklist, eventresult='Success')
```

Comment:
Interesting to understand if and why specifying starttime matters, given that the rule query period is 10 minutes anyways.

Reply:
not needed Ofer. As you mentioned, we are specifying query period in rule template.